Extension/src/Debugger/configurationProvider.ts

```diff
@@ -31,15 +31,15 @@ function isDebugLaunchStr(str: string): boolean {
  * Ensures that the selected configuration's preLaunchTask (if existent) is populated in the user's task.json.
  * Automatically starts debugging for "Build and Debug" configurations.
  */
-export class QuickPickConfigurationProvider implements vscode.DebugConfigurationProvider {
-    private underlyingProvider: CppConfigurationProvider;
+export class CppDebugConfigurationProvider implements vscode.DebugConfigurationProvider {
```

Comment:
Let's not rename all this stuff.  Can you change it back?

Naming a class "Underlying" isn't a very helpful choice.  "Underlying" only has meaning within the context that it's used.  A standalone class is not "underlying" anything.

We shouldn't use the abbreviation `dbg` for variable names.  In general abbreviations should be avoided, but in this case it also causes confusion with the `cppdbg` debug type which is an actual thing.

Reply:
the name underlying is actually currently existing, I removed it in the next PR, in refactoring the code.